### PR TITLE
Code Review revisions

### PR DESCRIPTION
**Pull request by BigAB**
As per code Review comments in #136 

Stylistic changes: resolving one push after if rather than in if and else blocks, use disabled attribute binding rather than `{{if}}` wrapping on elements. Removed some extra unneeded code.

Removing references to RefId hash maps and replacing with getters that do an indexOf loop. Removed the need for hashMaps and some init methods.

This also revealed that `contribution-month/monthly-client-projects-os-project.js` was pretty redundant/useless so I removed it and replaced it with a list of `MonthlyOSProject`s instead.

### CODE REVIEW

File: public/components/client-projects/client-projects.stache [hunk 1]
```diff
@@ -34,9 +34,9 @@
             {{#each monthlyOSProjects as monthlyOSProject}}
                 <input id="{{osProjectRef._id}}"
                        type="checkbox"
-                       {{#if monthlyClientProjectsOSProjects.has(monthlyOSProject) }}checked{{/if}}
+                       {{#if monthlyOSProjects.has(monthlyOSProject) }}checked{{/if}}
                        value="{{osProjectRef.value.name}}"
-                       ($click)="toggleUseProject(monthlyClientProjectsOSProjects, monthlyOSProject)"
+                       ($click)="toggleUseProject(monthlyOSProjects, monthlyOSProject)"
```
Comment (justinbmeyer) [inline public/components/client-projects/client-projects.stache]: Is this right?

File: public/components/os-projects/os-projects.js [hunk 2]
```diff
@@ -30,6 +29,7 @@ export const ViewModel = DefineMap.extend({
 
   // Derived properties
   creatingNewOSProject: {
+    type: 'boolean',
```
Comment (justinbmeyer) [inline public/components/os-projects/os-projects.js]: this doesn't do anything

File: public/lib/ref-list-utils.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export const indexOfRefForModel = (refList, model, refPropertyName = 'ref') => {
+  let index = -1;
+  refList.forEach( (ref, i) => { // eslint-disable-line consistent-return
+    if (ref[refPropertyName].id === model.id) {
```
Comment (justinbmeyer) [inline public/lib/ref-list-utils.js]: _id?

Comment (justinbmeyer) [inline public/lib/ref-list-utils.js]: Maybe add a fancy:

```
.index({id: 5})
```

File: public/models/contribution-month/monthly-os-project.js [hunk 4]
```diff
@@ -3,66 +3,54 @@ import DefineList from "can-define/list/";
...
+      } );
       return contributionMonth;
     },
     Type: ContributionMonth,
     serialize: false
   },
-  monthlyOSProjectIdMap: {
-    type: "any",
-    value: {}
+  has( osProject ){
```
Comment (justinbmeyer) [inline public/models/contribution-month/monthly-os-project.js]: create a tech debt issue to move this back into using a map, but a map regenerated via `getter` on .length change.

File: public/components/contributors/contributors_test.js [hunk 5]
```diff
@@ -139,9 +139,9 @@ QUnit.asyncTest('Remove an OS project to a monthly client project', function() {
             res(req.data);
         }
     });
-    const monthlyClientProjectsOSProjects = vm.contributionMonth.monthlyClientProjects[0].monthlyClientProjectsOSProjects;
-    vm.toggleUseProject(vm.contributionMonth, monthlyClientProjectsOSProjects, newMonthlyOSProject).then(() => {
-        QUnit.equal(vm.contributionMonth.monthlyClientProjects[0].monthlyClientProjectsOSProjects.length , 0);
+    const monthlyOSProjects = vm.contributionMonth.monthlyClientProjects[0].monthlyOSProjects;
+    vm.toggleUseProject(vm.contributionMonth, monthlyOSProjects, newMonthlyOSProject).then(() => {
+        QUnit.equal(vm.contributionMonth.monthlyClientProjects[0].monthlyOSProjects.length , 0);
```
Comment (justinbmeyer) [inline public/components/contributors/contributors_test.js]: Change to `monthlyClientProjectsOSRefs`

**Comment by BigAB** [comment]
Closing and replacing with #156 and #157

**Pull request closed without merging**